primary_beam_correction.py

```diff
@@ -0,0 +1,296 @@
+#!/usr/bin/env python
+import argparse
+import logging
+import numpy as np
+from astropy import units as u
+from astropy.io import fits
+from astropy import wcs
+from astropy import coordinates
+
+
+def read_fits(pathfits):
+    '''
+    This functions reads in the fits file
+    Input: Path to the fits file
+    Output: Fits file object
+    '''
+    fl = fits.open(pathfits)
+    images = fl[0]
+    return images
+
+
+def get_positions(path):
+    '''
+    This function reads the fits file and returns the sky coordinate
+    of the phase centre.
+    Input: path to the fits file
+    Output: Sky coordinate of the phase centre
+    '''
+    images = read_fits(path)
+    # Get pointing centre of observation
+    phase_centre_ra = images.header['OBSRA']
+    phase_centre_dec = images.header['OBSDEC']
+    # Convert to astropy.coordinates.SkyCoord object
+    phase_center = coordinates.SkyCoord(phase_centre_ra, phase_centre_dec, unit=(u.deg, u.deg))
+    # Get positions of other sources
+    source_pos = wcs.WCS(images.header)
+    return phase_center, source_pos
```

Comment:
This isn't strictly what you should do, Since 'OBSRA" and 'OBSDEC' define the pointing centre rather than the phase center - 99.9% of the time these are the same.

The better way to do this is to use the WCS throughout. Here is a better version of this function:
```
def get_positions(path):
    """
    This function reads the fits file and returns the sky coordinate
    of the phase centre.
    Input: path to the fits file
    Output: Sky coordinate of the phase centre
    """
    image_wcs = wcs.WCS(path)
    # Get pointing centre of observation
    phase_centre_ra = image_wcs.celestial.wcs.crval[0]
    phase_centre_dec = image_wcs.celestial.wcs.crval[1]
    # Convert to astropy.coordinates.SkyCoord object
    phase_center = coordinates.SkyCoord(phase_centre_ra, phase_centre_dec, unit=(u.deg, u.deg))
    return phase_center, image_wcs
```

Reply:
Got it thanks.